src/compute/src/metrics.rs

```diff
@@ -161,6 +164,11 @@ impl ComputeMetrics {
                 help: "The remaining seconds until replica expiration. Can go negative, can lag behind.",
                 var_labels: ["worker_id"],
             )),
+            collection_count: registry.register(metric!(
+                name: "mz_compute_collection_count",
+                help: "The number and hydration status of maintained compute collections.",
+                var_labels: ["worker_id", "type", "hydrated"],
```

Comment:
@materializekatrina I decided to add a "type" field with values such as (`user`, `system`, `log`, `transient`) because I imagine we want to filter collections for 0dt hydration tracking. We'll at least want to filter out `transient` collections (SELECTs, subscribes) as they can cause noise, and we may want to only look at `user` collections because the number of `system`/`log` collections can change between Mz versions.